cpu/src/lib.rs

```diff
@@ -466,6 +482,58 @@ where
     }
 }
 
+fn index_of_word(addr: u32) -> usize {
+    return ((addr & !3) - 1) as usize;
```

Comment:
Why `- 1`? Shouldn't this just be `addr & !3`?

Reply:
I thought I need to account for that array index starts from zero. There can be index out of bounds error if `addr & !3` is 4. But actually I think this function is not correct anyway. I'll fix it. Thanks!